app/Http/Controllers/PhotoController.php

```diff
@@ -30,16 +30,33 @@ public function store(Request $request, int $id)
         foreach($files as $file) {
             $savedPath = $file->store("{$id}", 'album_data');
 
-            // TODO: EXIF DATA EXTRACTION
+            // EXIF DATA EXTRACTION
+            $exif = @exif_read_data($file); // Return false if no data can be read
+
+            if ($exif && array_key_exists("GPSLatitude" ,$exif) && array_key_exists("GPSLatitudeRef" ,$exif)) {
+                $latitude = $this->gps($exif["GPSLatitude"], $exif['GPSLatitudeRef']);
+            }
+            else {
+                $latitude = "";
+            }
+
+            if ($exif && array_key_exists("GPSLongitude", $exif) && array_key_exists("GPSLongitudeRef", $exif)) {
+                $longitude = $this->gps($exif["GPSLongitude"], $exif['GPSLongitudeRef']);
+            }
+            else {
+                $longitude = "";
+            }
+
+            $dateTime = $exif && array_key_exists("DateTime", $exif) ? new DateTime($exif["DateTime"]) : new DateTime();
```

Comment:
Be careful ! EXIF data might have be edited manually to add an invalid date. I didn't find anything about `exif_read_data` validating the date.
`DateTime` return an exception when an invalid date is given so you might want to handle it. (<https://www.php.net/manual/en/datetime.construct.php>)

Reply:
Done in 7c7fab0b84936ae321e25ae253bf344f450f8720